Ginger/GingerCore/ALM/RQM/ExportToRQM.cs

```diff
@@ -800,9 +852,183 @@ public static string GetExportedIDString(string externalID, string searchString)
             }
             else
             {
-                Reporter.ToLog(eLogLevel.ERROR, "Could not export to RQM, External Items Fields values are missing");
+                Reporter.ToLog(eLogLevel.INFO, "Could not export to RQM, External Items Fields values are missing");
             }
             return propertiesList;
         }
+
+        public Dictionary<Guid, string> CreateNewALMDefects(Dictionary<Guid, Dictionary<string, string>> defectsForOpening, List<ExternalItemFieldBase> defectsFields, bool useREST)
+        {
+            LoginDTO loginData = new LoginDTO() { User = ALMCore.DefaultAlmConfig.ALMUserName, Password = ALMCore.DefaultAlmConfig.ALMPassword, Server = ALMCore.DefaultAlmConfig.ALMServerURL };
+
+            Dictionary<Guid, string> defectsOpeningResults = new Dictionary<Guid, string>();
+            Dictionary<string, List<string>> defectsBFs = new Dictionary<string, List<string>>();
+            DefectData newDefect = new DefectData();
+            foreach (KeyValuePair<Guid, Dictionary<string, string>> defectForOpening in defectsForOpening)
+            {
+
+                if (defectForOpening.Value.ContainsKey("Summary"))
+                {
+
+                    newDefect.summary = defectForOpening.Value["Summary"];//screenshots,ActivityGroupExternalID,ActivityExternalID,BFExternalID1,BFExternalID2
+                }
+                if (defectForOpening.Value.ContainsKey("description"))
+                {
+
+                    newDefect.description = defectForOpening.Value["description"];
+                }
+
+
+                AddEntityFieldValues(defectsFields, newDefect, "defect");
+                ResultInfo resultInfo;
+                RQMConnect.Instance.RQMRep.GetConection();
+
+                resultInfo = RQMConnect.Instance.RQMRep.CreateDefectNew(loginData, newDefect, ALMCore.DefaultAlmConfig.ALMServerURL, RQMCore.ALMProjectGuid, ALMCore.DefaultAlmConfig.ALMProjectName, RQMCore.ALMProjectGroupName, null);
+                if (resultInfo.IsSuccess)
+                {
+                    defectsOpeningResults.Add(defectForOpening.Key, newDefect.WorkItemId);
+                }
+            }
+            return defectsOpeningResults;
+        }
+
+        private void AddEntityFieldValues(List<ExternalItemFieldBase> fields, DefectData newDefect, string entityType)
+        {
+
+
+            foreach (ExternalItemFieldBase field in fields)
+            {
+                try
+                {
+                    //if ((field.ToUpdate || field.Mandatory))
+                    //{
+                        if (!string.IsNullOrEmpty(field.SelectedValue) && field.SelectedValue != "Unassigned")
+                        {
+                            if (field.ItemType.ToLower() == "string" && field.ID.Contains("filedAgainst"))
```

Comment:
add single if for field.ItemType == "string" condition and add switch case for all conditions

Reply:
Done